lib/wal/wal.c

```diff
@@ -622,7 +688,11 @@ wal_open(
     wal_fileset_version_set(wal->wfset, wal->version);
 
     if (rp->dur_intvl_ms != HSE_WAL_DUR_MS_DFLT)
-        wal->dur_ms = clamp_t(long, rp->dur_intvl_ms, HSE_WAL_DUR_MS_MIN, HSE_WAL_DUR_MS_MAX);
+        wal->dur_ms = clamp_t(uint32_t, rp->dur_intvl_ms, HSE_WAL_DUR_MS_MIN, HSE_WAL_DUR_MS_MAX);
```

Comment:
Could we make clamp_t just use the typeof(second arg) instead of explicitly passing in the type? @beckerg

Reply:
We copied clamp_t from the linux kernel so changing the interface could be confusing to someone.  As for the type coercion, I could see caller might want to coerce the result of all internal operations and the result to a type different than the type of the default value.  Changing the semantics of this interface would necessitate an audit of all use cases and socializing the new semantics, hence IMHO not worth the trouble...